cmake/CompilerFlagsHelpers.cmake

```diff
@@ -88,7 +88,6 @@ foreach(COMPILER_LANGUAGE ${SUPPORTED_COMPILER_LANGUAGE_LIST})
     set(CMAKE_${COMPILER_LANGUAGE}_VECTORIZE "")
     set(IGNORE_UNKNOWN_PRAGMA_FLAGS "-Wno-unknown-pragmas")
     set(CMAKE_${COMPILER_LANGUAGE}_GEN_NATIVE "-march=native")
-    set(CMAKE_${COMPILER_LANGUAGE}_IGNORE_WARNINGS "-Wno-deprecated-register -Wno-writable-strings")
```

Comment:
@alexsavulescu : I believe `-Wno-writable-strings` will be still needed? Also, need to change line ~106.

Reply:
No, it is also unsupported: 
```bash
icpc: command line warning #10148: option '-Wno-deprecated-register' not supported
icpc: command line warning #10148: option '-Wno-writable-strings' not supported
```

Reply:
Ah ok! I was thinking you removed because those errors are fixed!

But register deprecation can be removed now for Clang on line 106.

Reply:
Yup, just updated. Thanks